internal/service/appconfig/environment.go

```diff
@@ -3,333 +3,476 @@ package appconfig
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
-	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs/fwdiag"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	fwtypes "github.com/hashicorp/terraform-provider-aws/internal/framework/types"
 	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
-	"github.com/hashicorp/terraform-provider-aws/internal/verify"
 	"github.com/hashicorp/terraform-provider-aws/names"
 )
 
-// @SDKResource("aws_appconfig_environment", name="Environment")
+// @FrameworkResource
 // @Tags(identifierAttribute="arn")
-func ResourceEnvironment() *schema.Resource {
-	return &schema.Resource{
-		CreateWithoutTimeout: resourceEnvironmentCreate,
-		ReadWithoutTimeout:   resourceEnvironmentRead,
-		UpdateWithoutTimeout: resourceEnvironmentUpdate,
-		DeleteWithoutTimeout: resourceEnvironmentDelete,
-		Importer: &schema.ResourceImporter{
-			StateContext: schema.ImportStatePassthroughContext,
-		},
+func newResourceEnvironment(context.Context) (resource.ResourceWithConfigure, error) {
+	r := &resourceEnvironment{}
+	r.SetMigratedFromPluginSDK(true)
+
+	return r, nil
+}
+
+type resourceEnvironment struct {
+	framework.ResourceWithConfigure
+}
+
+func (r *resourceEnvironment) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "aws_appconfig_environment"
+}
 
-		Schema: map[string]*schema.Schema{
-			"application_id": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ForceNew:     true,
-				ValidateFunc: validation.StringMatch(regexp.MustCompile(`[a-z0-9]{4,7}`), ""),
+func (r *resourceEnvironment) Schema(ctx context.Context, request resource.SchemaRequest, response *resource.SchemaResponse) {
+	s := schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"application_id": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[a-z0-9]{4,7}$`),
+						"value must contain 4-7 lowercase letters or numbers",
+					),
+				},
 			},
-			"environment_id": {
-				Type:     schema.TypeString,
+			"arn": schema.StringAttribute{
 				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
 			},
-			"arn": {
-				Type:     schema.TypeString,
+			"description": schema.StringAttribute{
+				Optional: true,
 				Computed: true,
+				Default:  stringdefault.StaticString(""),
```

Comment:
Is this for Plugin-SDK backward compatibility?

Reply:
Yes. I'll add a comment